analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -236,6 +234,73 @@ public void Instance_InstanceMethodCallDoesNotClearFieldsOnOtherInstances(string
             validator.ValidateTag("StaticField", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
         }
 
+        [TestMethod]
+        public void Instance_InstanceMethodCallClearsFields_ConditionalAccess()
+        {
+            var code = $@"
+ObjectField = null;
+StaticObjectField = null;
+var otherInstance = new Sample();
+this?.InstanceMethod();
+Tag(""Field"", ObjectField);
+Tag(""StaticField"", StaticObjectField);
+";
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.Invocation);
+            validator.ValidateTag("Field", x => x.HasConstraint<ObjectConstraint>().Should().BeFalse());
+            validator.ValidateTag("StaticField", x => x.HasConstraint<ObjectConstraint>().Should().BeFalse());
+        }
+
+        [DataTestMethod]
+        [DataRow("(condition ? this : otherInstance).InstanceMethod();")]
+        [DataRow("(condition ? this : otherInstance).ExtensionMethod();")] // invocation with flow-capture and conversion on the receiver
+        public void Instance_InstanceMethodCallClearsFields_Ternary(string instanceCall)
```

Comment:
For all new test methods
```suggestion
        public void Instance_InstanceMethodCall_ClearsFields_Ternary(string instanceCall)
```